core/shared/src/main/scala/zio/prelude/NotSubType.scala

```diff
@@ -0,0 +1,11 @@
+package zio.prelude
+
+object NotSubType {
+  // Encoding for "A is not a subtype of B"
+  trait <:!<[A, B]
+
+  // Uses ambiguity to rule out the cases we're trying to exclude
+  implicit def nsub[A, B]: A <:!< B            = null
+  implicit def nsubAmbig1[A, B >: A]: A <:!< B = null
+  implicit def nsubAmbig2[A, B >: A]: A <:!< B = null
```

Comment:
In ZIO, we have a convention of implementing this "trick".
https://github.com/zio/zio/blob/master/core/shared/src/main/scala-2.12-2.13/zio/CanFail.scala
Do you think you could do it more in line with that? :pray: